File: Java/Inheritance/SubclassTest/payroll/Employee.java
package payroll;

//public class is visible outside of its package
public class Employee {

	private int id;
	int hours;
	protected float rate;
	private static int count;

	public Employee(int h, float r) {
		id = 101 + count++;
		hours = h;
		rate = r;
	}

	public Employee() {
		this(0, 0); //calling another constructor of this class
	}

	public int getId() { return id; }

	public int getHours() { return hours; }
	public void setHours(int value) { hours = value; }

	public float getRate() { return rate; }
	public void setRate(float value) { rate = value; }

	public double getIncome() {
		double income = hours * rate;
		int ot = hours - 180;
		if(ot > 0)
			income += 50 * ot;
		return income;
	}

	public static int countInstances() { return count; }
}

File: DotNet/Inheritance/DerivedClassTest/DemoApp/Payroll/Employee.cs
namespace Payroll
{
    class Employee
    {
        private int hours;
        private float rate;
        private static int count;

        public Employee(int h, float r)
        {
            hours = h;
            rate = r;
            count += 1;
        }

        public Employee() : this(0, 0) {}

        public int Hours
        {
            get { return hours; }
            set { hours = value; }
        }

        public float Rate
        {
            get { return rate; }
            set { rate = value; }
        }

        //overridable method
        public virtual double Income()
        {
            double income = hours * rate;
            int ot = hours - 180;
            if(ot > 0)
                income += 50 * ot;
            return income;
        }

        public static int CountInstances()
        {
            return count;
        }


    }
}